apps/mobile/src/app/activity/index.tsx

```diff
@@ -1,24 +1,36 @@
 import { FetchWrapper } from '@/components/loading';
-import { ActivityLayout, ActivityList, hasActivity } from '@/features/activity';
+import { Screen } from '@/components/screen/screen';
+import { ActivityListItem } from '@/features/activity';
 import { ActivityEmpty } from '@/features/activity/activity-empty';
+import { RefreshControl, useRefreshHandler } from '@/features/refresh-control/refresh-control';
 import { useTotalActivity } from '@/queries/activity/account-activity.query';
+import { t } from '@lingui/macro';
 
-import { Box } from '@leather.io/ui/native';
+import { OnChainActivity } from '@leather.io/models';
+import { Text } from '@leather.io/ui/native';
 
 export default function ActivityScreen() {
   const activity = useTotalActivity();
+  const { refreshing, onRefresh } = useRefreshHandler();
+  const pageTitle = t({ id: 'activity.header_title', message: 'All activity' });
 
   return (
-    <ActivityLayout>
+    <Screen>
+      <Screen.Header centerElement={<Text variant="label01">{pageTitle}</Text>} />
       <FetchWrapper data={activity}>
-        {hasActivity(activity) ? (
-          activity.state === 'success' && <ActivityList activity={activity.value} />
-        ) : (
-          <Box width="100%" justifyContent="center" alignItems="center">
-            <ActivityEmpty />
-          </Box>
+        {activity.state === 'success' && (
+          <Screen.List
+            refreshControl={<RefreshControl />}
+            ListHeaderComponent={<Screen.Title>{pageTitle}</Screen.Title>}
+            data={activity.value as OnChainActivity[]} // TODO: Unclear why was this cast. Needs clearing up.
```

Comment:
Sloppy implementation is the reason ✌️  [See thread here](https://github.com/leather-io/mono/pull/1404/commits/02bc117694fb87f84a80d03efd61e97f992da897#r2161804675)

Reply:
Thanks for clarifying. Added those before I took time off. We can log and refactor.

Reply:
Yeah we can leave it as is for now and then fix it easily after